sw/device/tests/sim_dv/i2c_device_tx_rx_test.c

```diff
@@ -71,6 +71,13 @@ static volatile const uint8_t kI2cByteCount = 0;
 static volatile bool tx_empty_irq_seen = false;
 static volatile bool cmd_complete_irq_seen = false;
 
+/**
+ * these variables store values based on kI2cIdx
+ */
+static uint32_t i2c_irq_fmt_watermark_id;
```

Comment:
You don't need another variable to hold this value, main's i2c_ctx has a datamember where it's stored for the test. Because it's an essential part of the i2c_ctx there, the configuration should set it there.

Reply:
I've moved watermark ID initialisation to function `void ottf_external_isr(void) `.
This function now uses a local variable `i2c_irq_fmt_watermark_id` in order to avoid code duplication.